app/src/main/java/co/condorlabs/customcomponents/customimageview/AppCompactTouchImageView.kt

```diff
@@ -0,0 +1,372 @@
+                savedTranslationOnY = currentTranslationOnY
+                mode = DRAG
+            }
+            MotionEvent.ACTION_UP -> {
+                onClick(event)
+                mode = NONE
+            }
+            MotionEvent.ACTION_POINTER_UP -> {
+                mode = NONE
+            }
+            MotionEvent.ACTION_POINTER_DOWN -> {
+                midPoint(midpoint, event)
+                mode = ZOOM
+            }
+            MotionEvent.ACTION_MOVE -> {
+                if (mode == DRAG && currentScale != defaultScale) {
+                    val currentDrawableWidth = drawableWidth * currentScale
+                    val currentDrawableHeight = drawableHeight * currentScale
+
+                    if (drawableWidth <= drawableHeight) {
+                        currentTranslationOnY = if (currentDrawableHeight <= viewHeight) {
+                            (viewHeight - currentDrawableHeight) / 2
+                        } else {
+                            min(
+                                0F,
+                                max(
+                                    viewHeight - currentDrawableHeight,
+                                    savedTranslationOnY + event.y - eventStartPoint.y
+                                )
+                            )
+                        }
+                        if (currentDrawableWidth > viewWidth) {
+                            currentTranslationOnX = min(
+                                0F,
+                                max(
+                                    viewWidth - currentDrawableWidth,
+                                    savedTranslationOnX + event.x - eventStartPoint.x
+                                )
+                            )
+                        }
+                    } else {
+                        currentTranslationOnX = if (currentDrawableWidth <= viewWidth) {
+                            (viewWidth - currentDrawableWidth) / 2
+                        } else {
+                            min(
+                                0F,
+                                max(
+                                    viewWidth - currentDrawableWidth,
+                                    savedTranslationOnX + event.x - eventStartPoint.x
+                                )
+                            )
+                        }
+
+                        if (currentDrawableHeight > viewHeight) {
+                            currentTranslationOnY = min(
+                                0F,
+                                max(
+                                    viewHeight - currentDrawableHeight,
+                                    savedTranslationOnY + event.y - eventStartPoint.y
+                                )
+                            )
+                        }
+                    }
+
+                    updateImageMatrix()
+                }
+            }
+        }
+
+        return true
```

Comment:
its possible the first if line 122 and else line 147 create 2 method o reduce the complexity

Reply:
sure